components/url_sanitizer/browser/url_sanitizer_service.cc

```diff
@@ -0,0 +1,160 @@
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/url_sanitizer/browser/url_sanitizer_service.h"
+
+#include <memory>
+#include <vector>
+
+#include "base/json/json_reader.h"
+#include "base/logging.h"
+#include "base/strings/string_split.h"
+#include "base/strings/string_util.h"
+#include "base/task/task_runner_util.h"
+#include "base/task/thread_pool.h"
+#include "base/values.h"
+#include "extensions/common/url_pattern.h"
+#include "url/gurl.h"
+
+namespace brave {
+
+namespace {
+bool CreateURLPatternSetFromValue(const base::Value* value,
+                                  extensions::URLPatternSet* result) {
+  if (!value || !value->is_list())
+    return false;
+  std::string error;
+  bool valid = result->Populate(
+      base::Value::AsListValue(*value),
+      URLPattern::SCHEME_HTTP | URLPattern::SCHEME_HTTPS, false, &error);
+  if (!valid) {
+    VLOG(1) << "Unable to create url pattern:" << error;
+  }
+  return valid;
+}
+
+bool CreateParamsList(const base::Value* value,
+                      base::flat_set<std::string>* params) {
+  if (!value->is_list() || !params)
+    return false;
+
+  for (const auto& it : value->GetList()) {
+    params->insert(it.GetString());
+  }
+  return true;
+}
+
+base::flat_set<std::unique_ptr<URLSanitizerService::MatchItem>> ParseFromJson(
+    const std::string& json) {
+  auto parsed_json = base::JSONReader::ReadAndReturnValueWithError(json);
+  if (!parsed_json.has_value() || !parsed_json.value().is_list()) {
+    VLOG(1) << "Error parsing feature JSON: " << parsed_json.error().message;
+    return {};
+  }
+  base::flat_set<std::unique_ptr<URLSanitizerService::MatchItem>> matchers;
+  for (const auto& it : parsed_json.value().GetList()) {
+    auto* include_list = it.FindListPath("include");
+    if (!include_list)
+      continue;
+    extensions::URLPatternSet include_matcher;
+    if (!CreateURLPatternSetFromValue(include_list, &include_matcher))
+      continue;
+    auto* params_list = it.FindListPath("params");
+    base::flat_set<std::string> params;
+    if (!CreateParamsList(params_list, &params))
+      continue;
+
+    extensions::URLPatternSet exclude_matcher;
+    CreateURLPatternSetFromValue(it.FindListPath("exclude"), &exclude_matcher);
+    auto item = std::make_unique<URLSanitizerService::MatchItem>(
+        std::move(include_matcher), std::move(exclude_matcher), params);
+
+    matchers.insert(std::move(item));
+  }
+
+  return matchers;
+}
+
+}  // namespace
+
+URLSanitizerService::URLSanitizerService() = default;
+
+URLSanitizerService::~URLSanitizerService() = default;
+
+URLSanitizerService::MatchItem::MatchItem() = default;
+URLSanitizerService::MatchItem::~MatchItem() = default;
+
+URLSanitizerService::MatchItem::MatchItem(extensions::URLPatternSet in,
+                                          extensions::URLPatternSet ex,
+                                          base::flat_set<std::string> prm) {
```

Comment:
Is it possible to have these members initialised in the initialiser list?

```
  : include(std::move(in)), exclude(std::move(ex)), params(std::move(prm))
```

Reply:
done